cpp/src/decisiontree/decisiontree.cuh

```diff
@@ -289,11 +290,6 @@ class DecisionTree {
       (std::numeric_limits<L>::is_integer) ? CRITERION::ENTROPY : CRITERION::MSE;
 
     validity_check(tree_params);
-    if (tree_params.n_bins > n_sampled_rows) {
```

Comment:
Why is this check removed? Is it moved to some place else

Reply:
It's not correct because the quantiles have already been computed at this stage, checked with @venkywonka on this. I don't necessarily see any reason to enforce nbins < nrows.